lib/_http_outgoing.js

```diff
@@ -382,7 +382,7 @@ OutgoingMessage.prototype.setHeader = function setHeader(name, value) {
 
 
 OutgoingMessage.prototype.getHeader = function getHeader(name) {
-  if (arguments.length < 1) {
+  if (typeof name !== 'string') {
```

Comment:
What do you mean? If someone does `getHeader(undefined)` for example it now throws a different error. An argument *was* supplied, now it's just the wrong type.